text/ProofModes.md

```diff
@@ -0,0 +1,118 @@
+# Proof Modes
+
+This CEP is about the infrastructure needed in order to have multiple 
+proof modes coexisting in a modular way (i.e. without syntactic or semantic conflicts)
+
+## Status quo (as per Coq 8.6)
+
+1. Tactic names are kernel names, i.e. unique, and can be accessed this way
+
+    ```
+    foo.tac arg1.
+    bar.tac arg1.
+    ```
+    
+1. Linking (not even requiring) a plugin loads into the *lexer* the tokens 
+   specific to the grammar implemented by the plugin
+   
+1. Linking (not even requiring) a plugin loads into the *parser* grammar the rules
+   specific to the plugin 
+   
```

Comment:
A slight technical point here is what to do when a plugin is "unloaded", that is to say, imagine the plugin successfully registers its tables, now when we go to the part of the document before the plugin was loaded, should these tables be dropped too?

With my current design they will, as the tables themselves are part of the parsing state, thus, we may need to be sure that these tables are registered even when the "already loaded" optimization is done.

The other alternative is to make these tables global, but again, it would be good if we could avoid that.